assets/src/stories-editor/components/template-inserter/index.js

```diff
@@ -24,6 +24,44 @@ import addTemplateIcon from '../../../../images/add-template.svg';
 import './edit.css';
 import { createSkeletonTemplate, maybeEnqueueFontStyle } from '../../helpers';
 
+class TemplatePreview extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			parentLoaded: false,
+		};
+	}
+
+	componentDidMount() {
+		if ( ! this.state.parentLoaded ) {
+			// Set timeout to cause a small latency between loading the templates, otherwise they all try to load instantly and cause a lag.
+			setTimeout( () => {
```

Comment:
@swissspidy Let me know if you have a better solution for this issue.

Basically, it looks like `react-window` and its lazy-loading is also not usable since it's assuming that each list item which needs to get lazy-loaded is being fetched. However, in our case there is nothing to fetch, it's rendering itself which takes time and we can't really add a promise for rendering and component.

Added this option which loads the templates with a small latency so it displays the inserter with spinners at first and then loads the templates with small time differences.

It loads at first just with a spinner and once loaded the first time, changes the state to load the actual preview.

Tried a bunch of different options (including injecting the previews after loading the frame) but this seemed to work most reliably.

Let me know if you have a better idea.

Reply:
At first glance I fear that using `setTimeout` is in fact the best solution here. I've seen similar patterns in the past too, which also indicates that this is the way to go.

Just a few minor notes / questions:

1. This should use the `withSafeTimeout` HOC so that timeouts are properly cleared when components are unmounted.
2. Maybe rename `parentLoaded` to `shouldLoad`
3. Is the timeout of `100` enough? Or could it even be `0` / omitted in order to just run during the next tick in the event loop?
4. Haven't tested yet, but how busy does the UI feel with all these spinners appearing for a short time? Should it perhaps just `return null` instead?

Reply:
1. 👍 
2. Yep, that's more accurate actually.
3. It seemed to be enough when testing, omitting or setting to `0` didn't work, however. We can test with some alternative values but `100` seemed to work well.
4. Well, the visible window is quite small, so not all the spinners are really visible, locally testing the spinners displayed just for a moment and then the first one was replaced already. One thing that we could do would be to also pass the index of the template to the `TemplatePreview` and display the spinner only if it's the first Template, this way we would show only one spinner but would still indicate that something is happening. Thoughts?